### PR TITLE
Support jupyterlab 3

### LINKED ISSUE #72: Support JupyterLab 3
It's worth a look to see which (if any) of the upstreams haven't been released to support lab3. Going to 3 would remove node/npm entirely from this repo for everything except `prettier`, which we can vendor, and would improve both development time and potentially CI.

## real dependencies

| extension      | status | link |
|----------------|--------|------|
| @jupyter-widgets/jupyterlab-manager | ready | https://pypi.org/project/jupyterlab-widgets/1.0.0/
| @pyviz/jupyterlab_pyviz | ready | https://pypi.org/project/pyviz-comms/2.0.1/
| jupyter-cytoscape | ready | https://pypi.org/project/ipycytoscape/1.2.0/
| qgrid2 | dead | ---

### what to do about qgrid
- alternatives
    - [ipyregulartable](https://github.com/jpmorganchase/ipyregulartable)
        - support merged, probably releasing _real soon now_
    - [wxyz-datagrid](https://github.com/deathbeds/wxyz/issues/42)
        - not started yet

## dev/docs dependencies
| extension      | status | link |
|----------------|--------|------|
| @deathbeds/jupyterlab-drawio-notebook@0.8.0-alpha1 | not started | https://github.com/deathbeds/jupyterlab-drawio/issues/11
| @deathbeds/jupyterlab-drawio-webpack@13.6.2-alpha1 | not started | https://github.com/deathbeds/jupyterlab-drawio/issues/11
| @deathbeds/jupyterlab-drawio@0.8.0-alpha1 | not started | https://github.com/deathbeds/jupyterlab-drawio/issues/11

### PR DESCRIPTION
**Pull request by zwelz3**
[Binder](https://mybinder.org/v2/gh/zwelz3/ipyradiant/gh-72-support-jupyterlab-3?urlpath=lab)

Update to JL3 and pin cytoscape to latest version

Closes #72

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ pip install ipyradiant
 Install the lab extensions that `ipyradiant` depends on.
 
 ```bash
-jupyter labextension install @jupyter-widgets/jupyterlab-manager jupyter-cytoscape @pyviz/jupyterlab_pyviz qgrid2
+jupyter labextension install @jupyter-widgets/jupyterlab-manager jupyter-cytoscape @pyviz/jupyterlab_pyviz
```
Comment (nrbgt) [inline README.md]: for lab 3, you won't do any of this: they're already part of the conda/pip packages

File: labextensions.txt [hunk 2]
```diff
@@ -4,4 +4,3 @@
 @jupyter-widgets/jupyterlab-manager
 @pyviz/jupyterlab_pyviz
 jupyter-cytoscape
-qgrid2
```
Comment (nrbgt) [inline labextensions.txt:7]: you'll be able to delete this file entirely

Comment (zwelz3) [reply labextensions.txt:7]: 1e6f88a

File: README.md (hunk 1 above)
Comment (zwelz3) [reply README.md]: 1e6f88a

File: postBuild [hunk 3]
```diff
@@ -10,8 +10,6 @@ python -m pip install -e . --no-cache-dir --no-deps
 
 # lab
 jupyter labextension list
-jupyter labextension install $(cat labextensions.txt) --no-build --debug
-jupyter labextension list
```
Comment (nrbgt) [inline postBuild:14]: won't need the `jupyter lab build` either :point_down:

Comment (zwelz3) [reply postBuild:14]: 613135e

**Comment by nrbgt** [comment]
ipywidgets 8 is coming soon... you'll probably want to pin that in your project, as well as `install_requires`, e.g. `ipywidgets >=7.6,<8` until we know for sure what the damage will be...

File: setup.cfg [hunk 4]
```diff
@@ -35,7 +35,7 @@ install_requires =
     datashader
     holoviews
     ipycytoscape >=1.0.3
-    ipywidgets
+    ipywidgets >=7.6,<8
```
Comment (nrbgt) [inline setup.cfg:38]: looks like your `ipyctyoscape` pin need updating :point_up: pyviz_comms, probably also needs an update... their stuff i think that pin is there to _avoid_ pulling the lab3 one

Comment (zwelz3) [reply setup.cfg:38]: Can I just remove the pyviz_comms dependency/pin altogether for now?

Comment (nrbgt) [reply setup.cfg:38]: dunno... not sure if it comes along with hv/ds... you'd have to check _their_ install_requires. If it's jupyter-only (possible) they probably won't, and you would want it.

**Comment by zwelz3** [comment]
Okay all green. Thanks for helping locate everything @nrbgt

**Pull request merged**